pydantic/types.py

```diff
@@ -549,7 +546,7 @@ def serialize(
             # hence we just use `secret_display`
             return _secret_display(value.get_secret_value())
         else:
-            return value
+            return value  # pragma: no cover
```

Comment:
Seems this line is not reachable because the function is used only for `json`

https://github.com/pydantic/pydantic/blob/c18947586b237c66113ebd1bef7465210ab45772/pydantic/types.py#L581-L585

Reply:
This should be reachable somehow, otherwise the conditional doesn't make sense, right?

Reply:
I think it should. but I don't know how TBH.
I was about to remove the condition but as I am not fully aware I've decided to keep the condition and ignore the coverage.

Reply:
I think a way to test this would be to create a new type: `class NewSecret(SecretStr)`, and override the `__get_pydantic_core_schema__`, and replace the value you want.